### PR TITLE
[Code Example] Add Brain Tumor Segmentation

### PR DESCRIPTION
**Pull request by innat**
Add multi-model brain tumor segmentation pipelines.

segmentation_slices (4)

### CODE REVIEW

**Comment by innat** [comment]
@divyashreepathihalli I've added necessary files. Please check.

File: scripts/tutobooks.py [hunk 1]
```diff
@@ -72,7 +72,7 @@
 from pathlib import Path
 
 TIMEOUT = 12 * 60 * 60  # 12 hours
-MAX_LOC = 350
+MAX_LOC = 500
```
Comment (divyashreepathihalli) [inline scripts/tutobooks.py]: Revert this change.

File: examples/vision/brain_tumor_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,838 @@
...
+    data = {"image": sample["image"], "label": sample["label"]}
+
+    pipeline = Compose(
+        [
+            ConvertToMultiChannelBasedOnBratsClasses(keys=["label"]),
+            NormalizeIntensity(keys=["image"], nonzero=True, channel_wise=True),
+        ]
+    )
+    result = pipeline(data, meta)
+    return result["image"], result["label"]
```
[the remark this answers is not in the document]
Comment (innat) [reply examples/vision/brain_tumor_segmentation.py:310]: /gemini

Its right that non of the applied transformation requires `meta` information. Its typically usefull for `spacing` and `orientation` level transformation. However, we like to keep it verbose for end user so that they can understand how to use `meta` data for their cause.

File: examples/vision/brain_tumor_segmentation.py [hunk 3]
```diff
@@ -0,0 +1,838 @@
...
+def test_transformation(sample):
+    data = {"image": sample["image"], "label": sample["label"]}
+    pipeline = Compose(
+        [
+            ConvertToMultiChannelBasedOnBratsClasses(keys=["label"]),
+            NormalizeIntensity(keys=["image"], nonzero=True, channel_wise=True),
+        ]
+    )
+    result = pipeline(data)
+    return result["image"], result["label"]
```
[the remark this answers is not in the document]
Comment (innat) [reply examples/vision/brain_tumor_segmentation.py]: /gemini

This is simply because to make things verbose and transferability

File: scripts/tutobooks.py (hunk 1 above)
Comment (innat) [reply scripts/tutobooks.py]: @divyashreepathihalli 
You mentioned to update this, [here](https://github.com/keras-team/keras-io/pull/2099#issuecomment-3829191527) because of [this](https://github.com/keras-team/keras-io/pull/2099#issuecomment-3106665471). Should I set minimum value to run this example?

Comment (divyashreepathihalli) [reply scripts/tutobooks.py]: Oh I suggested that for you to generate the files. Just for this example. Once you successfully did that, you need to change this back.

Comment (divyashreepathihalli) [reply scripts/tutobooks.py]: Oh I suggested that for you to generate the files. Just for this example. Once you successfully did that, you need to change this back.

**Review by divyashreepathihalli — APPROVED** [review]

**Comment by innat** [comment]
@divyashreepathihalli 
Thanks for your feedback. I see this [example](https://keras.io/examples/vision/brain_tumor_segmentation/) is online now. I like to ask

1. The example labels as "using keras 2", but keras 3 is used. Is there anything I can do from my side to fix it?
2. The example is categorized as **Other** section, whereas it should be in **3D** section. Again, is there anything I can do to fix this?
3. In the example, we download the dataset from kaggle with kagglehub, the output looks like this

```
0%| | 0.00/1.66G [00:00<?, ?B/s]

0%| | 1.00M/1.66G [00:00<12:12, 2.44MB/s]

0%|▏ | 3.00M/1.66G [00:00<04:26, 6.68MB/s]

0%|▍ | 8.00M/1.66G [00:00<01:40, 17.7MB/s]

1%|▌ | 13.0M/1.66G [00:00<01:07, 26.2MB/s]

1%|▉ | 19.0M/1.66G [00:00<00:52, 33.9MB/s]

1%|█▏ | 24.0M/1.66G [00:00<00:45, 38.5MB/s]
```

Also, output logs of `model.fit` suffers the same 

```
Epoch 1/4

20/20 ━━━━━━━━━━━━━━━━━━━━ 148s 462ms/step - dice: 0.1189 - dice_et: 0.0788 - dice_tc: 0.1141 - dice_wt: 0.1637 - loss: 1.6568

Epoch 2/4

20/20 ━━━━━━━━━━━━━━━━━━━━ 0s 459ms/step - dice: 0.2062 - dice_et: 0.1818 - dice_tc: 0.1996 - dice_wt: 0.2372 - loss: 1.4219

Epoch 2: Running inference...
Total patch 48: 0%| | 0/12 [00:00<?, ?it/s]

Total patch 48: 8%|█████▊ | 1/12 [01:25<15:36, 85.17s/it]

Total patch 48: 17%|███████████▋ | 2/12 [01:25<05:53, 35.38s/it]

Total patch 48: 25%|█████████████████▌ | 3/12 [01:26<02:55, 19.47s/it]

Total patch 48: 33%|███████████████████████▎ | 4/12 [01:26<01:36, 12.00s/it]

Total patch 48: 42%|█████████████████████████████▏ | 5/12 [01:27<00:55, 7.87s/it]

Total patch 48: 50%|███████████████████████████████████ | 6/12 [01:27<00:32, 5.38s/it]

Total patch 48: 58%|████████████████████████████████████████▊ | 7/12 [01:28<00:19, 3.80s/it]
```

tqdm loops got messy, can we fix it? Thanks again.

**Comment by divyashreepathihalli** [comment]
@innat  - you would need to change it here  https://github.com/keras-team/keras-io/blob/master/scripts/examples_master.py
With "keras_3": True.
Are you able to regenerate the files and stage it locally and test it? - if it works out - open a new PR.
